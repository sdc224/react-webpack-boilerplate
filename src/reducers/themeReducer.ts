import { PayloadAction } from "typesafe-actions";
import { ThemeActionTypes, ThemeState } from "../reduxTypes/themeTypes";

const initialState: ThemeState = {
	darkMode: false
};

const themeReducer = (
	state = initialState,
	action: PayloadAction<string, boolean>
): ThemeState => {
	switch (action.type) {
		case ThemeActionTypes.SET_DARK_MODE:
			return { ...state, darkMode: action.payload };

		default:
			return state;
	}
};

export default themeReducer;
