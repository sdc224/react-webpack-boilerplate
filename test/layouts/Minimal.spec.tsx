import React from "react";
import { RenderResult } from "@testing-library/react";
import { Minimal } from "@layouts/";
import { MemoryRouter } from "react-router-dom";
import renderWithProviders from "../helpers/renderWithProvider";
import { initialState } from "@reducers/themeReducer";
import MockComponent from "../mocks/MockComponent";

let documentBody: RenderResult;

describe("<Minimal />", () => {
	beforeEach(() => {
		// need to use Router as Topbar has Link
		documentBody = renderWithProviders(
			<MemoryRouter>
				<Minimal>
					<MockComponent />
				</Minimal>
			</MemoryRouter>,
			{ theme: { initialState } }
		);
	});
	it("has mock test", () => {
		const { getByText } = documentBody;
		expect(getByText(/Mock Component/i)).toBeInTheDocument();
	});
	it("matches snapshot", () => {
		const { baseElement } = documentBody;
		expect(baseElement).toMatchSnapshot();
	});
});
