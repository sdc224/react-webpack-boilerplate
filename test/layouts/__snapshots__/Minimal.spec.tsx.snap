// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Minimal /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-root-4"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-root-6 mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <a
            href="/"
          >
            <img
              alt="Logo"
              src="/images/logos/logo--white.svg"
            />
          </a>
        </div>
      </header>
      <main
        class="makeStyles-content-5"
      >
        <p>
          Mock Component
        </p>
      </main>
    </div>
  </div>
</body>
`;
